oauth2/src/test/java/com/linecorp/armeria/internal/client/auth/oauth2/MockOAuth2ClientCredentialsService.java

```diff
@@ -37,11 +40,19 @@
 
     @Post("/client/")
     @Consumes("application/x-www-form-urlencoded")
-    public HttpResponse handleTokenGet(
-            @Header("Authorization") Optional<String> auth,
-            @Param("grant_type") Optional<String> grantType,
-            @Param("scope") Optional<String> scope) {
+    public HttpResponse handleTokenGet(ServiceRequestContext ctx,
+                                       @Header("Authorization") Optional<String> auth,
+                                       @Param("grant_type") Optional<String> grantType,
+                                       @Param("scope") Optional<String> scope) {
+        final CompletableFuture<HttpResponse> future = new CompletableFuture<>();
+        // Intentionally delay execution to test concurrent token update scenario.
+        ctx.eventLoop().schedule(() -> future.complete(handleTokenGet(auth, grantType, scope)),
+                                 100L, TimeUnit.MILLISECONDS);
+        return HttpResponse.from(future);
```

Comment:
nit: Use `HttpResponse.delayed()`?
```suggestion
        // Intentionally delay execution to test concurrent token update scenario.
        return HttpResponse.delayed(handleTokenGet(auth, grantType, scope), Duration.ofMillis(100);
```